### ISSUE TITLE
Glances command/binary not in the PATH in v3.4.0.1 docker images

### ISSUE THREAD

**Issue by Antiz96** [nicolargo/glances#2406]
**Describe the bug**

It seems like the `glances` command/binary itself isn't in the path of the Alpine docker images which makes having to use `/venv/bin/python3 -m glances [options/arguments]` instead of simply `glances [options/arguments]` within the container.  
This, for instance, makes the ["How to protect your Dockerized server (or Web server) with a login/password" option 1 documentation](https://github.com/nicolargo/glances/blob/develop/docs/docker.rst#option-1) incorrect.

I'm willing to help with this but it's unclear to me if this behavior is intended and it's just the documentation that needs to be updated accordingly or if this is actually a bug in the docker images (see additional context).

**To Reproduce**
*Following [documentation](https://github.com/nicolargo/glances/blob/develop/docs/docker.rst#option-1)*
1. `docker exec -it glances_docker sh`
2. `glances -s --password`
3. "sh: glances: not found"

**Expected behavior**
- If the behavior described above is **not** intended: 
Being able to simply launch `glances [options/arguments]`, as the documentation suggests. 

- If the behavior described above is intended:
Having the documentation updated accordingly.

**Environement (please complete the following information)**
 - Operating System (lsb_release -a or OS name/version): `Up-to-date "latest-full" docker image running on a Debian 11 physical host server`
 - Glances & psutil versions: `Glances v3.4.0.1 with PsUtil v5.9.5`
 - How do you install Glances (Pypi package, script, package manager, source): `"latest-full" docker image`
 - Glances test (only available with Glances 3.1.7 or higher):

 ```
===============================================================================
Glances 3.4.0.1 (/app/glances/__init__.py)
Python 3.11.3 (/venv/bin/python3)
PsUtil 5.9.5 (/venv/lib/python3.11/site-packages/psutil/__init__.py)
===============================================================================
alert         [OK]    0.00001s 
[]
amps          [OK]    0.00004s 
[]
cloud         [OK]    0.00002s 
{}
connections   [OK]    0.00492s 
{'ESTABLISHED': 1,
 'LISTEN': 1,
 'SYN_RECV': 0,
 'SYN_SENT': 2,
 'initiated': 2,
 'net_connections_enabled': True,
 'nf_conntrack_count': 0.0,
 'nf_conntrack_enabled': True,
 'nf_conntrack_max': 262144.0,
 'nf_conntrack_percent': 0.0,
 'terminated': 2}
containers    [OK]    0.00450s key=name 
[{'Command': ['/bin/sh', '-c', '/venv/bin/python3 -m glances -C /etc/glances.conf $GLANCES_OPT'],
  'Created': '2023-05-18T09:53:14.607058301Z',
  'Id': 'f4b4aab85db1245c7085db1e96b03a2660bb1e9f961bc1d3e1ea5d7a1c03b60d',
  'Image': ['nicolargo/glances:latest-full'],
  'Status': 'running',
  'Uptime': 'an hour',
  'cpu_percent': 36.957202331390505,
  'engine': 'docker',
  'io_r': None,
  'io_w': None,
  'key': 'name',
  'memory_usage': 90673152,
  'name': 'glances',
  'network_rx': None,
  'network_tx': None}, ...]
core          [OK]    0.00047s 
{'log': 12, 'phys': 6}
cpu           [OK]    0.00024s 
{'cpucore': 12,
 'ctx_switches': 14173,
 'guest': 0.6,
 'guest_nice': 0.0,
 'idle': 96.7,
 'interrupts': 5070,
 'iowait': 0.0,
 'irq': 0.0,
 'nice': 0.0,
 'soft_interrupts': 2921,
 'softirq': 0.0,
 'steal': 0.0,
 'syscalls': 0,
 'system': 0.1,
 'time_since_update': 2.017988443374634,
 'total': 16.4,
 'user': 3.2}
diskio        [OK]    0.00033s key=disk_name 
[{'disk_name': 'loop0',
  'key': 'disk_name',
  'read_bytes': 0,
  'read_count': 0,
  'time_since_update': 2.018812417984009,
  'write_bytes': 0,
  'write_count': 0}, ...]
folders       [OK]    0.00002s 
[]
fs            [OK]    0.00028s key=mnt_point 
[{'device_name': '/dev/sdb3',
  'free': 104148774912,
  'fs_type': 'ext4',
  'key': 'mnt_point',
  'mnt_point': '/etc/resolv.conf',
  'percent': 8.2,
  'size': 113480257536,
  'used': 9314705408}, ...]
gpu           [OK]    0.00001s 
[]
help          [OK]    0.00000s 
None
ip            [OK]    0.00014s 
{'address': '***',
 'gateway': '***',
 'mask': '***',
 'mask_cidr': '***',
 'public_address': '***',
 'public_info_human': '***'}
irq           [OK]    0.00036s key=irq_line 
[{'irq_line': 'LOC', 'irq_rate': 4394, 'key': 'irq_line', 'time_since_update': 2.019179105758667}, ...]
load          [OK]    0.00002s 
{'cpucore': 12, 'min1': 0.32470703125, 'min15': 0.37646484375, 'min5': 0.38720703125}
mem           [OK]    0.00010s 
{'active': 2572554240,
 'available': 17779888128,
 'buffers': 294350848,
 'cached': 1095999488,
 'free': 17779888128,
 'inactive': 13430853632,
 'percent': 46.8,
 'shared': 41467904,
 'total': 33407754240,
 'used': 15627866112}
memswap       [OK]    0.00014s 
{'free': 4000313344,
 'percent': 0.0,
 'sin': 0,
 'sout': 0,
 'time_since_update': 2.6541929244995117,
 'total': 4000313344,
 'used': 0}
network       [OK]    0.00020s key=interface_name 
[{'alias': None,
  'cumulative_cx': 157400,
  'cumulative_rx': 78700,
  'cumulative_tx': 78700,
  'cx': 0,
  'interface_name': 'lo',
  'is_up': True,
  'key': 'interface_name',
  'rx': 0,
  'speed': 0,
  'time_since_update': 2.0130562782287598,
  'tx': 0}, ...]
now           [OK]    0.00001s 
'2023-05-18 10:56:33 UTC'
percpu        [OK]    0.00006s key=cpu_number 
[{'cpu_number': 0,
  'guest': 0.0,
  'guest_nice': 0.0,
  'idle': 16.0,
  'iowait': 0.0,
  'irq': 0.0,
  'key': 'cpu_number',
  'nice': 0.0,
  'softirq': 0.0,
  'steal': 0.0,
  'system': 0.0,
  'total': 84.0,
  'user': 0.0}, ...]
ports         [OK]    0.00013s 
[]
processcount  [OK]    0.04084s 
{'pid_max': 0, 'running': 2, 'sleeping': 217, 'thread': 521, 'total': 326}
processlist   [OK]    0.00013s key=pid 
[{'cmdline': ['/venv/bin/python3', '-m', 'glances', '-C', '/etc/glances.conf', '-w', '-u', 'rcandau', '--password'],
  'cpu_percent': 30.5,
  'cpu_times': pcputimes(user=1077.4, system=18.89, children_user=0.05, children_system=0.0, iowait=0.0),
  'gids': pgids(real=0, effective=0, saved=0),
  'io_counters': [0, 872448, 0, 872448, 1],
  'key': 'pid',
  'memory_info': pmem(rss=59691008, vms=69632000, shared=10735616, text=4096, lib=0, data=55144448, dirty=0),
  'memory_percent': 0.17867411131913308,
  'name': 'python3',
  'nice': 0,
  'num_threads': 3,
  'pid': 284383,
  'status': 'R',
  'time_since_update': 2.0699799060821533,
  'username': 'root'}, ...]
psutilversion [OK]    0.00002s 
(5, 9, 5)
quicklook     [OK]    0.00027s 
{'cpu': 16.4,
 'cpu_hz': 4800000000.0,
 'cpu_hz_current': 3111494000.0,
 'cpu_name': '11th Gen Intel(R) Core(TM) i5-11600 @ 2.80GHz',
 'mem': 46.8,
 'percpu': [{...}, {...}, {...}, {...}, {...}, {...}, {...}, {...}, {...}, {...}, {...}, {...}],
 'swap': 0.0}
raid          [OK]    0.00007s 
{}
sensors       [OK]    0.00001s key=label 
[{'critical': 105,
  'key': 'label',
  'label': 'acpitz 0',
  'type': 'temperature_core',
  'unit': 'C',
  'value': 27,
  'warning': 105}, ...]
smart         [OK]    0.02461s 
[]
system        [OK]    0.00000s 
{'hostname': 'f4b4aab85db1',
 'hr_name': 'Alpine Linux 3.18.0 64bit',
 'linux_distro': 'Alpine Linux 3.18.0',
 'os_name': 'Linux',
 'os_version': '5.15.107-2-pve',
 'platform': '64bit'}
uptime        [OK]    0.00010s 
{'seconds': 90979}
wifi          [OK]    0.00002s 
[]
===============================================================================
Total time to update all stats: 0.08020s
===============================================================================
 ```

**Additional context**

As I said in the description, I'm unsure if this is a bug in the latest Alpine images or if it's the new intended behavior and it's just the documentation that hasn't been updated accordingly yet. If it's the latter, I'm sorry for the incorrect bug report and I'm willing to submit a PR containing the updated documentation if it helps :smiley:
If it's the former, I "solved" it on my side by creating the `/usr/bin/glances` file containing the following:
```
#!/bin/sh

/venv/bin/python3 -m glances $@
```
If you like this solution I could make a PR to integrate that into the Dockefile as well if it helps!

**Comment by RazCrimson** [comment]
@Antiz96 
I was the one who did the docker file changes.
My bad, I should have updated the PATH env var instead of using the venv in the docker CMD.

If you are interested in contributing, go ahead and make a PR with a fix :)

**Comment by Antiz96** [comment]
@RazCrimson Sure, I'll look into that.
Any preferred solution in mind?

From what I can see, the glances binary used to be a python script stored in `/usr/local/bin/` (the below is picked from a `v3.2.3` glances docker image):
`cat /usr/local/bin/glances`
```
#!/usr/bin/python3
# -*- coding: utf-8 -*-
import re
import sys
from glances import main
if __name__ == '__main__':
    sys.argv[0] = re.sub(r'(-script\.pyw|\.exe)?$', '', sys.argv[0])
    sys.exit(main())
 ```
Are you looking for something similar? To be honest, I'm unsure how to update it properly to make it work with glances' new version and the now used python virtual env. Not that much of a python guy myself :/
I assume the simple *(but somewhat hacky?)* `/venv/bin/python3 -m glances $@` in `/usr/local/bin/glances` solution described in my initial post won't do?

**Comment by RazCrimson** [comment]
@Antiz96 

Just a small change, please prepend `/venv/bin` to the PATH environment variable too

**Comment by Antiz96** [comment]
@RazCrimson Alright, I'll work on that over the week-end ;)

**Comment by Antiz96** [comment]
@RazCrimson By the way, here's a bit of additional context about how I found that bug: https://github.com/nicolargo/glances/issues/2410#issuecomment-1555823605

TLDR: The algorithm used by glances for encrypting passwords seems to have changes between `v3.3.X` and `v3.4.X`, forcing users to regenerate their passwords once again in `v3.4.X` in order to make the switch to the new expected algorithm. By trying to do so (`glances -s --password`), I saw that `glances` wasn't in the path anymore.

Anyway, I'll come back with a PR as soon as I can.
